arch/x86/x86.c

```diff
@@ -14,6 +14,22 @@ ZLIB_INTERNAL int x86_cpu_has_sse2;
 ZLIB_INTERNAL int x86_cpu_has_sse42;
 ZLIB_INTERNAL int x86_cpu_has_pclmulqdq;
 
+#ifdef _MSC_VER
+#include <intrin.h>
+#define CPU_PROCINFO_AND_FEATUREBITS 1
+
+void ZLIB_INTERNAL x86_check_features(void) {
+	unsigned int registers[4];
+	__cpuid(registers, CPU_PROCINFO_AND_FEATUREBITS);
+
+	unsigned int ecx = registers[2];
+	unsigned int edx = registers[3];
+
+	x86_cpu_has_sse2 = edx & 0x4000000;
```

Comment:
Knowing that the registers are ecx/edx etc is actually very useful for readability, so I'd prefer it stays like this or gets documented clearly. I would think the compiler will be able to optimize out the extra variable assignments here in any case.